typescript/phoenix-sdk/src/utils/market.ts

```diff
@@ -628,3 +633,299 @@ export function getMarketExpectedOutAmount({
+    [
+      trader.toBuffer(),
+      TOKEN_PROGRAM_ID.toBuffer(),
+      marketData.header.baseParams.mintKey.toBuffer(),
+    ],
+    ASSOCIATED_TOKEN_PROGRAM_ID
+  )[0];
+  const quoteAccount = PublicKey.findProgramAddressSync(
+    [
+      trader.toBuffer(),
+      TOKEN_PROGRAM_ID.toBuffer(),
+      marketData.header.quoteParams.mintKey.toBuffer(),
+    ],
+    ASSOCIATED_TOKEN_PROGRAM_ID
+  )[0];
+
+  const evictSeatAccounts = {
+    phoenixProgram: PROGRAM_ID,
+    logAuthority,
+    market,
+    seatManager,
+    seatDepositCollector,
+    baseMint: marketData.header.baseParams.mintKey,
+    quoteMint: marketData.header.quoteParams.mintKey,
+    baseVault: marketData.header.baseParams.vaultKey,
+    quoteVault: marketData.header.quoteParams.vaultKey,
+    associatedTokenAccountProgram: ASSOCIATED_TOKEN_PROGRAM_ID,
+    signer,
+    trader,
+    seat,
+    baseAccount,
+    quoteAccount,
+    baseAccountBackup: baseTokenAccountBackup ?? PublicKey.default,
+    quoteAccountBackup: quoteTokenAccountBackup ?? PublicKey.default
+  }
+
+  return createEvictSeatInstruction(evictSeatAccounts);
+}
+
+/**
+ * Checks if the given trader has a seat on the given market
+ * If not, return claim seat instructions
+ * @param connection The connection to the cluster
+ * @param market The market's address
+ * @param trader The trader's address
+ */
+export async function confirmOrCreateClaimSeatIxs(connection: Connection, market: PublicKey, trader: PublicKey): Promise<TransactionInstruction[]> {
+  const seat = getSeatAddress(market, trader);
+
+  const instructions: TransactionInstruction[] = [];
+
+  const seatAccountInfo = await connection.getAccountInfo(seat);
+  if ((seatAccountInfo === null) || (seatAccountInfo.data.length == 0)) {
+    instructions.push(getClaimSeatIx(market, trader));
+  }
+
+  return instructions;
+}
+
+// getPlaceLimitOrderSingleIx
+// accounts input: market, trader, 
+// params for orderpacket: side, priceInTicks, numBaseLots, selfTradeBehavior?, matchLimit?, clientOrderId?, useOnlyDepositedFunds?
+
+/**
+ * Returns an instruction to place a limit order on a market
+ * Does not check for existence of ATAs or seat (no network calls)
+ * Useful for existing market makers with existing associated token accounts for the market's base and mint tokens
+ * @param market The market's address
+ * @param marketData The market's data, containing base and quote mint information
+ * @param trader The trader's address
+ * @param orderPacket The order packet to place
+ */
+export function getPlaceLimitOrderSingleIx(market: PublicKey, marketData: MarketData, trader: PublicKey, orderPacket: OrderPacket): TransactionInstruction {
+  const seatManager = getSeatManagerAddress(market);
+  const seatDepositCollector = getSeatDepositCollectorAddress(market);
+  const seat = getSeatAddress(market, trader);
+  const logAuthority = getLogAuthorityAddress();
+
+  const baseAccount = PublicKey.findProgramAddressSync(
```

Comment:
Why do we use the sync version here?

Reply:
The regular `findProgramAddress` is marked as deprecated, with a note to use the sync version instead. Should we ignore that?